backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/log/ModularContractLogDAO.java

```diff
@@ -105,19 +109,19 @@ public ModularContractLogEntryId create(@NonNull final LogEntryCreateRequest req
 			log.setC_Invoice_Candidate_ID(invoiceCandidateId.getRepoId());
 		}
 
-		final Money amount = request.getAmount();
-		if (amount != null)
-		{
-			log.setAmount(amount.toBigDecimal());
-			log.setC_Currency_ID(amount.getCurrencyId().getRepoId());
-		}
+		Optional.ofNullable(request.getAmount())
```

Comment:
hmmm, if you would like to go the Optionals way, 
IMHO request.getAmount() shall return Optional.